frontend/file/src/main/java/com/virtuslab/gitmachete/frontend/file/MacheteFileWriter.java

```diff
@@ -0,0 +1,64 @@
+package com.virtuslab.gitmachete.frontend.file;
+
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.nio.file.Path;
+
+import com.intellij.openapi.application.ReadAction;
+import com.intellij.openapi.vfs.VfsUtilCore;
+import com.intellij.openapi.vfs.VirtualFileManager;
+import io.vavr.collection.List;
+import io.vavr.control.Option;
+import io.vavr.control.Try;
+import lombok.CustomLog;
+import lombok.val;
+
+import com.virtuslab.branchlayout.api.BranchLayout;
+import com.virtuslab.branchlayout.api.BranchLayoutException;
+import com.virtuslab.branchlayout.api.readwrite.IBranchLayoutWriter;
+import com.virtuslab.branchlayout.impl.readwrite.BranchLayoutFileUtils;
+import com.virtuslab.qual.guieffect.UIThreadUnsafe;
+
+@CustomLog
+public class MacheteFileWriter {
+
+  @UIThreadUnsafe
+  @SuppressWarnings("regexp") // to allow for `synchronized`
+  public synchronized void writeBranchLayout(Path path, BranchLayout branchLayout, boolean backupOldFile,
+      IBranchLayoutWriter branchLayoutWriter)
+      throws BranchLayoutException {
+    LOG.debug(() -> "Writing branch layout to (${path}), branchLayout = ${branchLayout}, backupOldFile = ${backupOldFile}");
+    val macheteVFile = VirtualFileManager.getInstance().findFileByNioPath(path);
+
+    try {
+      if (macheteVFile != null) {
+        if (backupOldFile) {
+          val parentDir = macheteVFile.getParent();
+          assert parentDir != null : "Can't get parent directory of branch layout file";
+
+          val backupFileName = macheteVFile.getName() + "~";
+          val backupVFile = ReadAction.compute(() -> Option.of(parentDir.findChild(backupFileName)));
```

Comment:
Same, aren't we supposed to be within `WriteAction` here? hence `ReadAction.compute` is probably redundant (if not erroneous)